Added polymorphic associations notifiable and device_compatible

tsov opened the pull request:
The CoreUser object now has notifications and devices.
Added tests for core user spec

Commit:
added polymorphic association device_compatible to core users, tests attached

Commit:
added polymorphic association notifiable to core users, tests attached

Commit:
separated tests in core user spec

Commit:
Merge branch 'development' of github.com:brogrammers/eventhub-api into devices-notifications-user-association

Conflicts:
	app/models/business_user.rb
	app/models/core_user.rb
	spec/models/core_user_spec.rb

foFox commented on app/models/core_user.rb:
```diff
@@ -2,6 +2,8 @@ class CoreUser < ActiveRecord::Base
 
+  has_many :devices, :as => :device_compatible, :dependent => :destroy
+  has_many :notifications, :as => :notifiable, :dependent => :destroy
   has_many :identities, :as => :identifiable, :dependent => :destroy
```
I just spotted that, can you add validations, so user must have at least one identity before save ?

tsov replied:
Yeah man, will do

Commit:
added validation so a user needs at least one identity + test cases

tsov commented:
Ok, I updated the branch, I added some test cases and had to mess with the fixtures, so some of the users have the association set up on fixture load in. I'll explain tomorrow.
It should be good to go though
